accepted/2024/observable_collection.md

```diff
@@ -0,0 +1,201 @@
+# Enabling Batch Events for ObservableCollection
+
+**Owner** [Immo Landwerth](https://github.com/terrajobst) | [Eirik Tsarpalis](https://github.com/eiriktsarpalis)
+
+The .NET platform has infrastructure to enable *view models* that is, an object
+model that UI can be bound to such that when the data changes the UI refreshes
+automatically.
+
+One of the core building blocks for this is `INotifyCollectionChanged`, with
+it's canonical implementation of `ObservableCollection<T>`.
+
+While the shape of `INotifyCollectionChanged.CollectionChanged` always supported
+expressing batched notifications, `ObservableCollection<T>` only raises
+notifications for individual items (except for the `Reset` event, which tells UI
+to update everything).
+
+In fact, `ObservableCollection<T>` has no APIs like `AddRange` which could allow
+bulk operations. In the past we tried to add those but then we learned that some
+of the .NET platform pieces took a dependency on the fact that
+`ObservableCollection<T>` raises events for individual items and crash when the
+event args represent more than one item, specifically WPF.
+
+From a UI standpoint, bulk operations are desirable because they allow for UI
+controls to be more performant when the data changes. Other UI frameworks that
+support view models, such as Avalonia, Uno, and MAUI have expressed interest in
+seeing this change. In fact, the [tracking issue][issue] is eight years old and
+has over four hundred comments, with several folks expressing frustration that
+this issue hasn't been addressed yet.
+
+The goal of this document is to explore ways to address this request.
+
+## Requirements
+
+### Goals
+
+* Enable bulk operations for `Add`, `Remove`, `Insert`, and `Replace` that
+  result in a single notification for view models.
+
+### Non-Goals
+
+* Support bulk notifications for discontinuous ranges in the collection
+
+## Stakeholders and Reviewers
+
+* Platform components
+    - Libraries team
+    - WinForms team
+    - WPF team
+    - WinUI team
```

Comment:
We should also mention UWP, since it supports modern .NET as well now.
Things should just work on it anyway, since it's basically the same as WinUI 3.